ScanController.ipf

```diff
@@ -374,105 +374,255 @@ end
-//	loadbooleanJSONfromkeys(numkeys,numdestinations)
-//
-//	// load calc wave configuration
-//	// keys are: wavenames:calc, record_waves:calc, plot_waves:calc, scripts:calc
-//	textkeys = "wave_names:calc,scripts:calc"
-//	numkeys = "record_waves:calc,plot_waves:calc"
-//	textdestinations = "sc_CalcWaveNames,sc_CalcScripts"
-//	numdestinations = "sc_CalcRecord,sc_CalcPlot"
-//	loadtextJSONfromkeys(textkeys,textdestinations)
-//	loadbooleanJSONfromkeys(numkeys,numdestinations)
-//
-//	// load print checkbox settings
-////	sc_PrintRaw = booltonum(stringfromlist(0,extractJSONvalues(getJSONkeyindex("print_to_history",t_tokentext),children="raw"),","))
-////	sc_PrintCalc = booltonum(stringfromlist(0,extractJSONvalues(getJSONkeyindex("print_to_history",t_tokentext),children="calc"),","))
-//
-//	// load log string
-////	sc_LogStr = unescapeJSONstr(stringfromlist(0,extractJSONvalues(getJSONkeyindex("log_string",t_tokentext)),","),notwave=1)
+	// waves
+	loadtextarrayintowave(getvalueTOML(TOMLstr,"waves:raw"),"sc_RawWaveNames")
+	loadtextarrayintowave(getvalueTOML(TOMLstr,"waves:calc"),"sc_CalcWaveNames")
+	
+	// record checkboxes
+	loadbooleanarrayintowave(getvalueTOML(TOMLstr,"checkboxes.record:raw"),"sc_RawRecord")
+	loadbooleanarrayintowave(getvalueTOML(TOMLstr,"checkboxes.record:calc"),"sc_CalcRecord")
+	
+	// plot checkboxes
+	loadbooleanarrayintowave(getvalueTOML(TOMLstr,"checkboxes.plot:raw"),"sc_RawPlot")
+	loadbooleanarrayintowave(getvalueTOML(TOMLstr,"checkboxes.plot:calc"),"sc_CalcPlot")
+	
+	// async checkboxes
+	loadbooleanarrayintowave(getvalueTOML(TOMLstr,"checkboxes.asybc:async"),"sc_measAsync")
+	
+	// print_to_history
+	loadbooleanintovariable(getvalueTOML(TOMLstr,"checkboxes.history:raw"),"sc_PrintRaw")
+	loadbooleanintovariable(getvalueTOML(TOMLstr,"checkboxes.history:calc"),"sc_PrintCalc")
+	
+	// scripts
+	loadtextarrayintowave(getvalueTOML(TOMLstr,"scripts:raw"),"sc_RawScripts")
+	loadtextarrayintowave(getvalueTOML(TOMLstr,"scripts:calc"),"sc_CalcScripts")
+	
+	// executable string to get logs
+	loadtextintostring(getvalueTOML(TOMLstr,"logstring:logstr"),"sc_Logstr")
+	
+	//filenum
+	loadnumintovariable(getvalueTOML(TOMLstr,"filenum:num"),"sc_filenum")
+	
+	// reload ScanController window
+	//sc_rebuildwindow()
+end
+
+function/s getvalueTOML(TOMLstr,key)
+	// returns the value associated with key
+	// returns an empty string is key is not valid
+	string TOMLstr,key
+	variable key_length,index,old_index,i=0, val_start, val_end
+	string str
+	
+	key_length = itemsinlist(key,":")
+	
+	// search TOMLstr for key
+	old_index = 0
+	for(i=0;i<key_length;i+=1)
+		if(i<key_length-1)
+			str = "["+stringfromlist(i,key,":")+"]"
+		else
+			str = stringfromlist(i,key,":")
+		endif
+		index = strsearch(TOMLstr,str,old_index)
+		if(index==-1 || index<old_index)
+			return ""
+		endif
+		old_index=index+strlen(str)
+	endfor
+	val_start = index+strlen(str)+1
+	val_end = strsearch(TOMLstr,num2char(13),val_start) // look for \r
```

Comment:
I think this is going to break if you happen to be using Windows. The line endings are OS specific. Look for "\r" or "\n" might be right?

Reply:
I checked! if you don't specific /T=<term> in freadline igor adds \r by default. So it should be consistent across platforms.

Reply:
Good work, Igor.